lib/assert.js

```diff
@@ -185,6 +185,50 @@ function isArguments(object) {
   return Object.prototype.toString.call(object) == '[object Arguments]';
 }
 
+function compareMap(a, b, strict) {
+  // check size of the maps
+  if (a.size !== b.size)
+    return false;
+
+  if (!compareMapKeys(a, b, strict))
+    return false;
+
+  return compareMapKeys(b, a, strict);
+}
+
+function compareMapKeys(a, b, strict) {
+  // iterate over all keys and check if they exist in the other
+  for (let kvs of a) {
+    var key = kvs[0];
+    var val = kvs[1];
+    if (!b.has(key)) {
+      return false;
+    }
+
+    if (!_deepEqual(val, b.get(key), strict))
+      return false;
+  }
+  return true;
+}
+
+function compareSet(a, b, strict) {
+  if (a.size !== b.size)
+    return false;
+
+  if (!compareSetItems(a, b, strict))
+    return false;
+
+  return compareSetItems(b, a, strict);
+}
+
+function compareSetItems(a, b, strict) {
+  for (let item of a) {
+    if (!b.has(item))
```

Comment:
> Strictly speaking sets are unordered

Strictly correct, but note that unlike ES5 Objects where iteration is in "whatever" order, iteration over Sets and Maps is defined in ES2015 to occur in "insertion order".

Despite this, while verifying order may be more "correct" I disagree that this is useful in practice though. If you want to verify order you can convert the set/map.keys() to an array and deepEqual those. For the common usecase of sets/maps order shouldn't matter.

> It should check the same index for sets, not sure about maps.
both sets and Maps should exhibit the same behaviour.